lesson2/homework/helloworld/src/main/java/Animal.java

```diff
@@ -95,16 +98,18 @@ public void grow() {
 
 public class Farm {
 
+    private ArrayList<Animal> animals;
+
     public Farm(){
-        ArrayList animals = new ArrayList();
+        animals = new ArrayList<>();
     }
-    public void addAnimal(Object n_animal){
+    public void addAnimal(Animal n_animal){
         animals.add(n_animal);
     }
-    public Object getAnimal(int index){
-        return animals(index);
+    public Animal getAnimal(int index){
```

Comment:
Same thing here, make your function return a specific `Animal` object instead of a generic `Object`.